### ISSUE TITLE
Error 0x80092013 cloning a repo

### ISSUE THREAD

**Issue by jmporter34** [git-for-windows/git#1505]
- [x ] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

2.16.1.windows.4, 64 bit

```
$ git --version --build-options

git version 2.16.1.windows.4
cpu: x86_64
built from commit: ef6d451bbfef86a529ebf12620289e0f15a93d8e
sizeof-long: 4

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

Windows 7 Pro, SP1, 64 bit

```
$ cmd.exe /c ver

Microsoft Windows [Version 6.1.7601]

```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

Mostly the defaults.  I chose the option to use both Bash and Cmd prompt.  

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: Notepad++
Path Option: Cmd
SSH Option: OpenSSH
CURL Option: WinSSL
CRLF Option: CRLFAlways
Bash Terminal Option: ConHost
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled

```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

The problem only happens on my personal laptop on my home WiFi.  I can successfully clone the same
 repo using my corporate laptop at work, and using the same corporate laptop at home on my WiFi.  I'm not able to test my personal laptop at work (without taking a chance on getting fired...).
Also, I was able to successfully clone a different repo from a different hosting service with the same personal laptop on home WiFi that I'm having the trouble with here.


 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Bash and Windows Cmd

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

git clone https://github.com/NordicSemiconductor/IOS-nRF-Toolbox.git

```
 - What did you expect to occur after running these commands?

successful cloning of the repo to my hard drive

 - What actually happened instead?

Cloning into 'IOS-nRF-Toolbox'...
fatal: unable to access 'https://github.com/NordicSemiconductor/IOS-nRF-Toolbox.
git/': schannel: next InitializeSecurityContext failed: Unknown error (0x8009201
3) - The revocation function was unable to check revocation because the revocati
on server was offline.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

see above.  I tried a couple other repos on GitHub, all failed the same way.
e.g. https://github.com/brechtsanders/xlsxio.git

Note:
------
I first discussed this issue with "shiftkey" after replying to his response to 
https://github.com/desktop/desktop/issues/2414
He asked me to create a new issue and attach my Git config:

$ git config --global --list
winupdater.recentlyseenversion=2.16.1.windows.4
user.name=John Porter
user.email=jmporter34@gmail.com
merge.tool=p4merge
mergetool.p4merge.path=C:/Program Files/Perforce/p4merge.exe
mergetool.prompt=false
diff.tool=p4merge
difftool.p4merge.path=C:/Program Files/Perforce/p4merge.exe
difftool.prompt=false
http.schannel.checkrevoke=false

$ git config --system -l
http.sslcainfo=C:/Program Files/Git/mingw64/ssl/certs/ca-bundle.crt
http.sslbackend=schannel
diff.astextplain.textconv=astextplain
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.smudge=git-lfs smudge -- %f
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
credential.helper=manager
core.editor='C:\Program Files\Notepad++\notepad++.exe' -multiInst -notabbar -nos
ession -noPlugin

**Comment by dscho** [comment]
> schannel: next InitializeSecurityContext failed: Unknown error (0x80092013) - The revocation function was unable to check revocation because the revocation server was offline.

I saw some similar bug reports and while I do not yet understand everything in enough detail, it seems that it often happens either because of proxy issues or because the Secure Channel tries to reach a server that is unreachable due to a VPN being offline.

Does this sound like something that could be happening in your setup?

To work around these issues, @shiftkey provided PR #1450: you can set the config variable `http.schannel.checkRevoke` to `false` and it should work.

Does this help your use case?

**Comment by jmporter34** [comment]
Hi, 
I had already communicated with @shiftkey before opening this issue.  I had tried his "checkrevoke=false" suggestion, but the same failure still occurred after setting it and restarting Bash. 
 
I do usually have NordVPN running, but for a test I configured it not to run at startup, re-booted Windows and checked that neither the NordVPN App nor Process showed up in Task Manager.  Then I checked that I still had "checkrevoke=false" in my global config.  Then I tried cloning the repo again, but got the same failure again.
I also verified that I don't have any proxy configured.

From my reading about this and some similar issues, the thinking around this seems to be that it's related to corporate networks / proxy settings.  It's interesting that I'm only seeing the problem on my personal laptop at home,

**Comment by dscho** [comment]
@jmporter34 any chance you have a BitDefender firewall on your personal laptop? I found [this ticket](https://github.com/rust-lang/cargo/issues/1689#issuecomment-145215276) that seems to suggest BitDefender causing all kinds of problems with SSL.

**Comment by jmporter34** [comment]
Yes I do use BitDefender on my personal laptop.  I turned its firewall off and tried to clone the repo again but got the same error.  Then I turned its firewall back on and tried to clone the repo again, and this time I was able to get the repo.  I tried the other repos that had been failing and they all worked too now.  I had literally tried this a hundred times without any success before toggling the BitDefender firewall, so it has to have been that.  It's interesting that I had to toggle the firewall off and then back on before things started working though; I would have expected turning it off and leaving it off to do the trick.
Thank you for all the help !

**Comment by shiftkey** [comment]
Thanks @jmporter34 and @dscho for doing the debugging while I was asleep, and identifying how BitDefender relates to the problem.

I believe that we're invoking curl correctly in #1450, and like the original problem external components are meddling with the SSL operation, so I'm not sure what else we can do in Git for Windows to handle this...

**Comment by dscho** [comment]
I think that we'll want to add this to the FAQ. Maybe we can even work with BitDefender to identify and fix the underlying issue?

@jmporter34 do you have a way to contact BitDefender's technical support?

I am interested to see this resolved because I'd like to switch Git for Windows' default to Secure Channel some time this year...

**Comment by jmporter34** [comment]
I'll open a ticket with BitDefender support and keep you posted.

**Comment by dscho** [comment]
@jmporter34 thank you!

**Comment by jmporter34** [comment]
Here's what I got back from BitDefender support (first, I had to update to the 2018 version):
-----------------------------------------------------------------------------------------------------

First task:
Open the main interface of Bitdefender 2018, then click on the menu button from the upper left corner of the window.
Click on Protection from the left sidebar, then click on View Features.
Use the switch to turn the Firewall ON in case it was off, then click on the gear icon under Firewall.
Select the Network Adapters tab, and select Home/Office next to the names of your network adapters
Now switch to the Settings tab, then click on Edit stealth settings, and turn off this feature for your network adapters.

Second task:
Open the main interface of Bitdefender 2018, then click on the menu button from the upper left corner of the window.
Click on Protection from the left sidebar, then click on View Features from the right side of the window.
Click on the gear icon under the Firewall module.
In the Rules tab, click on Add Rule.
In the window that opens first enable the option called “Apply this rule to all applications” under the Settings tab.  NOTE: I BELIEVE THAT THIS STEP WOULD HAVE ALLOWED ALL APPLICATIONS TO BYPASS THE FIREWALL, SO INSTEAD I CREATED A CUSTOM RULE THAT ALLOWS ONLY "C:\Program Files\Git\bin\git.exe" TO BYPASS THE FIREWALL.
Click on OK to save your custom network exception rule.

Third task:
Open the main interface of Bitdefender 2018 then click on the menu button from the upper left corner.
Select Protection from the left sidebar, then click on View Features from the right side of the window.
Click on the gear icon under Advanced Threat Defense.
Click on “Add applications to the whitelist”
Select the process for the application you wish to exclude (a .exe file normally) then click on OK.

Fourth task:
Restart the computer and check if the issue persists.

After following the instructions:
-----------------------------------
I restarted my system, then tried cloning a few repos that had failed initially (the failures at the beginning of this issue report).  Everything worked fine.  However, I'm not sure whether any of this actually solved the problem. 

I say that because after all the initial failures a few days ago, I had toggled the BitDefender Firewall off and back on, then everything began working, as I described a few messages back.  The failure never re-appeared, even after I restarted mys system, turned the Firewall off and on, disabled and re-enabled BitDefender, ran some virus scans, basically everything I could think of.  I was never able to reproduce the issue again.  

So, all that is proven by following the instructions for configuring the Firewall, restarting and testing a few repos, is that it didn't break anything.  The original issue may be solved, or it may just not have reappeared yet.  If I had a way to reliably reproduce the 0x80092013 issue, I could say conclusively that configuring the Firewall fixed it.  But I can't reproduce it, so I can't say that ...
